eta/core/image.py

```diff
@@ -368,6 +372,51 @@ def merge_labels(self, image_labels):
         self.add_image_attributes(image_labels.attrs)
         self.add_objects(image_labels.objects)
 
+    @property
+    def has_schema(self):
+        '''Returns True/False whether the container has a schema.'''
+        return self.schema is not None
+
+    def set_schema_and_filter(self, schema):
+        ''' Filters this label object by the provided schema
+        Args:
+            schema (ImageLabelsSchema): schema to set and filter by
+        '''
+        self.schema = schema
+        self._filter_by_schema()
+
+    def _filter_by_schema(self):
+        if self.has_schema:
+            self._filter_image_attributes()
+            self._filter_objects()
+
+    def _filter_image_attributes(self):
+        attr_container = AttributeContainer()
+        for attr in self.attrs:
+            if self.schema.has_image_attribute(attr.name):
+                attr_container.add(attr)
+        self.attrs = attr_container
+
+    def _filter_objects(self):
```

Comment:
Re @benjaminpkane's comment: `AttributeContainer` has `AttributeContainerSchema`, so it would make sense to move the `AttributeContainer` filtering logic to `AttributeContainer.set_schema_and_filter()`.

Unfortunately I was lazy and there is no `DetectedObjectContainerSchema`, so looping over the objects would be necessary here.

Reply:
Thought that there might be a better place to put this logic so it is more reusable.